### PR TITLE
feat: prompt for missing variables on renku init

### LINKED ISSUE #1126: Read variables list when using renku init
**Is your feature request related to a problem? Please describe.**
Users need to know the variable list required by a template to initialize a new project

**Describe the solution you'd like**
Once SwissDataScienceCenter/renku-project-template#68 is merged, we can 
* [x] Print the variables list in `--print-manifest` (renamed as `--list-templates`)
* [x] Remove `--description` (currently not used) since there is no need to have it as a special variable
* [x] Rename `--template-variables` to `--variables`
* [ ] [bonus]: when some variables are required but not specified, interactively ask them to the user -- we may move this to a new issue

**Describe alternatives you've considered**
We could scan all the files in the template and try to find all the jinja variables, but that requires checking out all the files and it doesn't provide a description.

**Additional context**
Listing the variables is needed also for implementing #862

### PR DESCRIPTION
**Pull request by lorenzo-cavazzi**
# Description
The parameters required by the template selected during `renku init` were defaulted to `""` if not provided with the command through the `--parameter` options. This led to a bad UX, especially when the user selected a template after reading the manifest.

This PR implements a comparison between the provided parameters and the required ones. If they don't match:
* Missing parameters will be prompted.
* Useless parameters will be removed and a notification will be printed.

Fixes #1126

### CODE REVIEW

File: renku/cli/init.py [hunk 1]
```diff
@@ -388,6 +389,30 @@ def init(
...
+                f'({template_variables[key]})'
+            ),
+            default='',
+            show_default=False
+        )
+        parameter[key] = value
+    useless_variables = input_parameters_keys - template_variables_keys
+    if (len(useless_variables) > 0):
+        click.echo(
+            'These parameters are not required by the template: {}'.format(
```
Comment (Panaetius) [inline renku/cli/init.py:409]:
```suggestion
            INFO + 'These parameters are not used by the template and were ignored:\n\t {}'.format(
```

together with `from renku.core.commands.echo import INFO`

File: renku/cli/init.py [hunk 2]
```diff
@@ -388,6 +389,30 @@ def init(
...
+                ', '.join(useless_variables)
```
Comment (Panaetius) [inline renku/cli/init.py:410]:
```suggestion
                '\n\t'.join(useless_variables)
```

File: renku/cli/init.py (hunk 1 above)
Comment (lorenzo-cavazzi) [reply renku/cli/init.py]: oh nice! I didn't know about `INFO`. Thanks!

File: renku/cli/init.py (hunk 2 above)
Comment (lorenzo-cavazzi) [reply renku/cli/init.py]: done

**Pull request merged**